File: Programmers/Inho/C#/Level01/로또의 최고 순위와 최저 순위.cs
using System;

public class Solution
{
    public int[] solution(int[] lottos, int[] win_nums)
    {
        int rank = lottos.Length + 1;
        int zeroNum = 0;
        int sameNum = 0;
        int maxRank = 0;
        int minRank = 0;
        foreach (int a in lottos)
        {
            if (a == 0)
                zeroNum++;
            foreach (int b in win_nums)
            {
                if (a == b)
                    sameNum++;
            }
        }
        minRank = rank - sameNum;
        maxRank = minRank - zeroNum;
        if (sameNum <= 1)
            minRank = 6;
        if (sameNum + zeroNum <= 1)
            maxRank = 6;
        return new int[] { maxRank, minRank };
    }
}

File: Programmers/Inho/C++/Level01/로또의 최고 순위와 최저 순위.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    vector<int> answer;
    int correct = 0;
    int zero = 0;
    for(int i = 0 ;i<lottos.size();i++){
        for(int j = 0; j<win_nums.size();j++){
            if(lottos[i]==win_nums[j])
                correct++;
        }
        if(lottos[i]==0)
            zero++;
    }
    zero +=correct;
    if(zero == 0)
        zero = 1;
    if(correct == 0)
        correct = 1;
    answer.push_back(7-zero);
    answer.push_back(7-correct);
    return answer;
}